### DISCUSSION TITLE
OpenAI Whisper Modern GUI

### DISCUSSION THREAD

**Discussion by rudymohammadbali** [openai/whisper#1561] [Show and tell]
# Preview
![Screenshot 2023-08-07 104238](https://github.com/openai/whisper/assets/63475761/f8c6e8de-0035-434b-a897-a15fd22c2d60)

# Link to the project
https://github.com/rudymohammadbali/OpenAI-Whisper-GUI

**Comment by IrvingMor** [comment]
This GUI is great! And it also helped me use GPU since it was giving me a ton of errors in my attempt to use it. Just a question is there a possibility to add timestamps to the output that saves? I used the format_timestamp on write in my little attempt but I don't know if its a possibility in the code you've shared (I'm a programing noob in general srry for the question) And thanks for sharing!

**Reply by rudymohammadbali** [reply]
@IrvingMor 
This features is already in the app, when saving the output choose file type as (.srt, .vtt or .tsv) the timestamps will be added. another way is to click (Show Time) checkbox it will do the same job.

**Reply by IrvingMor** [reply]
Huh, I see. I seem to be having trouble with both versions where the last one just never processes the file and the first one just finishes but doesn't let me save the transcript. But none of them show an error in console. So i cant test this yet. Any clue as to why this might be happening? Also thanks for the reply!

**Reply by rudymohammadbali** [reply]
@IrvingMor I didn't understand your issue. however the app cloud have some bugs. I'm working on modern and fully production ready a version of Whisper GUI. it will release soon

**Comment by 7gxycn08** [comment]
Translate function is missing you can place this at the beginning of the def run_transcribe and it should translate with no issues.
```
def run_transcribe(self, file_path, model, language, task, device, show_time):
        if task == "translate":
            notification = Notification(
                master=self,
                text="Task has started. Please wait!",
                cl_btn=False,
                progress_bar=True)
            notification.show_message()
            self.start_button.configure(state="disabled")
            subprocess.call(fr"whisper {file_path} --output_format srt --language {language.title()}  --model {model}")
            self.start_button.configure(state="normal")
            notification.hide_message()
            notification = Notification(master=self, text="Task complete!")
            notification.show_message()
            self.after(5000, notification.hide_message)
            return
```

**Comment by benuildo** [comment]
I tried it but I get this error. Any ideas?
`RuntimeError: don't know how to restore data location of torch.storage.UntypedStorage (tagged with gpu)`
